### PR TITLE
Posts & Pages: New design for pages cells

### LINKED ISSUE #21820: Posts & Pages: Update Pages cell design

### PR DESCRIPTION
**Pull request by kean**
Fixes #21820, removes the Undo button, and update the Templates design.

Known issues:

- The "Published/Edited" part is missing from the date label (should be done together with the Posts)
- The list doesn't update when you create a new page (hope it's not a regression; I put it on the list)
- The custom homepage is not displayed https://github.com/wordpress-mobile/WordPress-iOS/issues/21841

To test:

- Create a new site with a FSE theme and verify that the Homepage template cell is displayed and has an updated design
- Verify that "regular" pages have an updated design

## Regression Notes
1. Potential unintended areas of impact: Pages List
2. What I did to test those areas of impact (or what existing automated tests I relied on): manual
3. What automated tests I added (or what prevented me from doing so): n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Pages/PageListItemViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+private func makeBadgesString(for page: Page) -> String {
+    var badges: [String] = []
+    if page.isSiteHomepage {
+        badges.append(Strings.badgeHomepage)
+    } else if page.isSitePostsPage {
+        badges.append(Strings.badgePosts)
+    }
+    if let displayDate = page.displayDate() {
+        badges.append(displayDate.capitalized)
```
Comment (momo-ozawa) [inline WordPress/Classes/ViewRelated/Pages/PageListItemViewModel.swift:48]: As you mentioned already, we'll need to prepend dates with "Published", "Edited", or "Scheduled". When we do, I think we can drop .capitalized as well as the .capitalizeFirstWord in PostListItemViewModel

**Review by momo-ozawa — APPROVED** [review]
Looking good! I think we should use the same top / bottom margins in the homepage template cell and the "normal" pages cells - wdyt? cc: @osullivanchris 

Current | top / bottom margin = 12.0 
-- | --
 |

**Comment by osullivanchris** [comment]
> Looking good! I think we should use the same top / bottom margins in the homepage template cell and the "normal" pages cells - wdyt? cc: @osullivanchris


Sounds good. Some borders are indented some not. Am I right in saying the full-width ones are for the homepage template cell and for the end of the list? If so I think that works...by the way I don't really know the rules iOS has for indented or full-width so I just went with what looked best to me.

**Comment by kean** [comment]
Good catch, I totally missed that on a simulator. I fixed both the margins and the separator insets:

Screenshot 2023-10-23 at 9 18 09 AM

File: WordPress/Classes/ViewRelated/Pages/PageListItemViewModel.swift (hunk 1 above)
Comment (kean) [reply WordPress/Classes/ViewRelated/Pages/PageListItemViewModel.swift:48]: Let's remove it once we add these labels.

**Pull request merged**